frontend/src/utilities/panic.ts

```diff
@@ -0,0 +1,49 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function panicProxy(module: any) {
+	const proxyHandler = {
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		get(target: any, propKey: any, receiver: any) {
+			const interceptedFunction = target[propKey];
+
+			// TODO: Figure out how to wrap constructors, instead of skipping them for now
+			const isClass = typeof interceptedFunction === "function" && /^\s*class\s+/.test(interceptedFunction.toString());
+			if (isClass) return interceptedFunction;
+
+			const targetValue = Reflect.get(target, propKey, receiver);
+			if (typeof targetValue === "function") {
+				// eslint-disable-next-line @typescript-eslint/no-explicit-any, func-names
+				return function (...args: any) {
+					let returned;
+					try {
+						// @ts-expect-error
+						returned = targetValue.apply(this, args);
```

Comment:
If this is the only branch of this block that can return, why not use `return targetValue.apply(this, args);`

Reply:
We're catching the exception, so we still need to return the function call's value.